### ISSUE TITLE
推理时 解析model output变量

### ISSUE THREAD

**Issue by stjack01** [KimMeen/Time-LLM#142]
Hi @kwuking ，
 
你好，我希望在推理中解析出Target变量的值，从下面的代码，我理解的model的output的维度是[batch, pred_len+label_len, variables]. 就是第3个维度是模型的全部变量（X,Y）的维度。 但是我实际打印output的维度时发现，outputs.shape= [24,1280,1], 似乎模型每次输出的output只包含一个变量的结果？ 

features =M

如果training和infer的时候使用features=M， 我需要自己判断outputs的第三个维度具体对应哪个变量？ 不知道我的理解对吗？

            if args.output_attention:
                    outputs = model(batch_x, batch_x_mark, dec_inp, batch_y_mark)[0]
                else:
                    outputs = model(batch_x, batch_x_mark, dec_inp, batch_y_mark)

                **f_dim = -1 if args.features == 'MS' else 0**
                outputs = outputs[:, -args.pred_len:, f_dim:]
                batch_y = batch_y[:, -args.pred_len:, f_dim:]
                loss = criterion(outputs, batch_y)
                train_loss.append(loss.item())

我同时检查了batch_x, batch_x_mark，batch_y_mark 这3个变量的输入维度，发现都是[24,96,1] , 似乎是因为data_loader.py的如下代码？ 每次get_item只返回一个变量 （feat_id = 变量的序号）？

        seq_x = self.data_x[s_begin:s_end, feat_id:feat_id + 1]
        seq_y = self.data_y[r_begin:r_end, feat_id:feat_id + 1]
        seq_x_mark = self.data_stamp[s_begin:s_end]
        seq_y_mark = self.data_stamp[r_begin:r_end]

我的问题是，outputs = model(batch_x, batch_x_mark, dec_inp, batch_y_mark) 可以实现同时输入全部变量吗？ 假设有10个var， 预测长度=32，需要怎么做才能得到 10*32 的output？我应该怎么解析出Target变量的输出呢？ 非常感谢

![modelparams](https://github.com/user-attachments/assets/19b10403-8842-4216-bae9-fe3ed6067021)

**Comment by USTCdyf** [comment]
请问你解决了吗

**Comment by stjack01** [comment]
@USTCdyf , not yet

**Comment by leimu3407** [comment]
I alse meet this problem，everything refers to that “features” is “S”.  No matter what the "features"  is ?

Renewal

In my test，feat_id will change. Different features are used, but only one.

![image](https://github.com/user-attachments/assets/9e5b9420-53fb-49cf-8b27-9f3df40d68b9)

**Comment by XiaoGang-git** [comment]
> Hi @kwuking ，
> 
> 你好，我希望在推理中解析出Target变量的值，从下面的代码，我理解的model的output的维度是[batch, pred_len+label_len, variables]. 就是第3个维度是模型的全部变量（X,Y）的维度。 但是我实际打印output的维度时发现，outputs.shape= [24,1280,1], 似乎模型每次输出的output只包含一个变量的结果？
> 
> features =M
> 
> 如果training和infer的时候使用features=M， 我需要自己判断outputs的第三个维度具体对应哪个变量？ 不知道我的理解对吗？
> 
> ```
>         if args.output_attention:
>                 outputs = model(batch_x, batch_x_mark, dec_inp, batch_y_mark)[0]
>             else:
>                 outputs = model(batch_x, batch_x_mark, dec_inp, batch_y_mark)
> 
>             **f_dim = -1 if args.features == 'MS' else 0**
>             outputs = outputs[:, -args.pred_len:, f_dim:]
>             batch_y = batch_y[:, -args.pred_len:, f_dim:]
>             loss = criterion(outputs, batch_y)
>             train_loss.append(loss.item())
> ```
> 
> 我同时检查了batch_x, batch_x_mark，batch_y_mark 这3个变量的输入维度，发现都是[24,96,1] , 似乎是因为data_loader.py的如下代码？ 每次get_item只返回一个变量 （feat_id = 变量的序号）？
> 
> ```
>     seq_x = self.data_x[s_begin:s_end, feat_id:feat_id + 1]
>     seq_y = self.data_y[r_begin:r_end, feat_id:feat_id + 1]
>     seq_x_mark = self.data_stamp[s_begin:s_end]
>     seq_y_mark = self.data_stamp[r_begin:r_end]
> ```
> 
> 我的问题是，outputs = model(batch_x, batch_x_mark, dec_inp, batch_y_mark) 可以实现同时输入全部变量吗？ 假设有10个var， 预测长度=32，需要怎么做才能得到 10*32 的output？我应该怎么解析出Target变量的输出呢？ 非常感谢
> 
> ![modelparams](https://private-user-images.githubusercontent.com/140493448/365643715-19b10403-8842-4216-bae9-fe3ed6067021.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzNTY3OTgsIm5iZiI6MTczNTM1NjQ5OCwicGF0aCI6Ii8xNDA0OTM0NDgvMzY1NjQzNzE1LTE5YjEwNDAzLTg4NDItNDIxNi1iYWU5LWZlM2VkNjA2NzAyMS5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyOFQwMzI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yM2IzYzMzZWYxOGI0MTdiMWRhMjJhMDJjYmRmNjA3NTMyNjk4NjU2YjQ1OThjNTNlYjZiMjQ2ODE3MTczMTJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.eKiNFBh9c-NWwnIOZRj-OJToMuZn5P1DwPozDXpAbRk)

你好，我也遇到和你一样的问题，请问你后面有想到解决办法吗？另外代码中的test函数你能跑通吗？

**Comment by zhouzige** [comment]
请问解决了吗？我也是这样的问题
